src/lightning/fabric/wrappers.py

```diff
@@ -401,7 +401,12 @@ def _capture_compile_kwargs(compile_fn: Callable) -> Callable:
     # PyTorch will resolve this in the future: https://github.com/pytorch/pytorch/issues/116575
 
     @wraps(compile_fn)
-    def _capture(model: Any, **kwargs: Any) -> Any:
+    def _capture(*args: Any, **kwargs: Any) -> Any:
+        if not args:
+            # torch.compile is being applied as a decorator
+            return compile_fn(*args, **kwargs)
+
+        model = args[0]
```

Comment:
Your code works. I would also refactor to be more explicit about compiling functions that are not models

```python
    def _capture(*args: Any, **kwargs: Any) -> Any:
        if not args:
            # torch.compile is being applied as a decorator
            return compile_fn(*args, **kwargs)

        model = args[0]
        if not isinstance(model, nn.Module):
            # compiling something else
            return compile_fn(*args, **kwargs)
        
        compiled_model = compile_fn(model, **kwargs)
        compiled_model._compile_kwargs = deepcopy(kwargs)
        return compiled_model
```

Reply:
in that case, wouldn't we want to combine the two into branches into one?
```
if not args or not isinstance(args[0], nn.Module): 
    return ...
```

Reply:
Yeah